logger.go

```diff
@@ -41,37 +31,33 @@ func LogMetadata(c *cli.Context) error {
 }
 
 func setupLogging(c *cli.Context) error {
-	logger := newZapLogger(appName, c.Bool("debug"), usesProductionLoggingConfig(c))
-	c.Context.Value(loggerContextKey{}).(*atomic.Value).Store(logger)
-	return nil
+	log, err := newZapLogger(appName, c.Int(newLogLevelFlag().Name), usesProductionLoggingConfig(c))
+	c.Context = logr.NewContext(c.Context, log)
+	return err
 }
 
 func usesProductionLoggingConfig(c *cli.Context) bool {
-	return strings.EqualFold("JSON", c.String("log-format"))
+	return strings.EqualFold("JSON", c.String(newLogFormatFlag().Name))
 }
 
-func newZapLogger(name string, debug bool, useProductionConfig bool) logr.Logger {
+func newZapLogger(name string, verbosityLevel int, useProductionConfig bool) (logr.Logger, error) {
 	cfg := zap.NewDevelopmentConfig()
 	cfg.EncoderConfig.ConsoleSeparator = " | "
 	if useProductionConfig {
 		cfg = zap.NewProductionConfig()
 	}
-	if debug {
-		// Zap's levels get more verbose as the number gets smaller,
-		// bug logr's level increases with greater numbers.
-		cfg.Level = zap.NewAtomicLevelAt(zapcore.Level(-2)) // max logger.V(2)
-	} else {
-		cfg.Level = zap.NewAtomicLevelAt(zapcore.InfoLevel)
-	}
+	// Zap's levels get more verbose as the number gets smaller,
+	// bug logr's level increases with greater numbers.
+	cfg.Level = zap.NewAtomicLevelAt(zapcore.Level(verbosityLevel * -1))
 	z, err := cfg.Build()
 	zap.ReplaceGlobals(z)
```

Comment:
Something that existed before, but we probably shouldn't use `z` before we check the error